test/cppsuite/src/main/database_operation.cpp

```diff
@@ -124,6 +124,20 @@ database_operation::populate(
     logger::log_msg(LOG_INFO, "Populate: finished.");
 }
 
+void
+database_operation::background_compact_operation(thread_worker *tc)
+{
+    logger::log_msg(
+      LOG_ERROR, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+    /* This needs to be executed only once in the workload. */
+    if (tc->running()) {
+        const std::string compact_cfg(
+          "background=true,free_space_target=" + std::to_string(tc->free_space_target_mb) + "MB");
+        testutil_check(tc->session->compact(tc->session.get(), nullptr, compact_cfg.c_str()));
```

Comment:
This is for my interest and not a PR blocker. It looks like `compact` is serving two purposes now: 1) to perform a compaction, and 2) to enable background config. Is there a reason we haven't moved 2) into its own `compact_reconfigure` call?

Reply:
In a few words, it was fewer code changes to re-use an existing function and extend its configuration items. Happy to have a chat if you want to know more!